public/js/analytics/colorCurve.js

```diff
@@ -0,0 +1,141 @@
+function GetColorCat(type, colors) {
+  if (type.toLowerCase().includes('land')) {
+    return 'l';
+  }
+  if (colors.length === 0) {
+    return 'c';
+  }
+  if (colors.length > 1) {
+    return 'm';
+  }
+  if (colors.length === 1) {
+    switch (colors[0]) {
+      case 'W':
+        return 'w';
+      case 'U':
+        return 'u';
+      case 'B':
+        return 'b';
+      case 'R':
+        return 'r';
+      case 'G':
+        return 'g';
+      case 'C':
+      default:
+        return 'c';
+    }
+  }
+}
+
+onmessage = (e) => {
+  if (!e) return;
+  const cards = e.data;
+  const curve = {
+    white: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    blue: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    black: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    red: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    green: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    colorless: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    multi: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    total: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+  };
+
+  for (const card of cards) {
+    let category;
+    switch (GetColorCat(card.details.type, card.colors)) {
+      case 'w':
+        category = curve.white;
+        break;
+      case 'u':
+        category = curve.blue;
+        break;
+      case 'b':
+        category = curve.black;
+        break;
+      case 'r':
+        category = curve.red;
+        break;
+      case 'g':
+        category = curve.green;
+        break;
+      case 'm':
+        category = curve.multi;
+        break;
+      case 'c':
+        category = curve.colorless;
+        break;
+      default:
+        break;
+    }
+    // const asfan = card.asfan || 15 / cards.length;
+    const asfan = 1;
+    if (category) {
+      if (card.cmc >= 9) {
```

Comment:
```suggestion
      if ((card.cmc !== undefined ? card.cmc : card.details.cmc) >= 9) {
```
Or you could make this a utility function and replace throughout this file